src/angular-multi-select-utils.js

```diff
@@ -49,7 +49,9 @@ angular_multi_select_utils.factory('angularMultiSelectUtils', [
 			 * array or undefined.
 			 */
 			if (typeof(str) === 'string') {
-				return str.split(",").map(s => s.replace(/^\s+|\s+$/g, ''));
+				return str.split(",")
+					.map(s => s.replace(/^\s+|\s+$/g, ''))
+					.filter(s => s.length > 0);
```

Comment:
The best thing to do here is to, indeed as you say, remove the empty elements from the element in such a way it ends up being a balanced array.

A quick and dirty solution would be to check `if (str === "") return undefined;`.

If you feel brave try to do it the first way, else just fill a bug and I'll patch it ASAP.